Automattic/jetpack
# Classic Theme Helper: Add Restaurant Menus Custom Post Type to Classic Theme Helper Package

Fixes https://github.com/Automattic/vulcan/issues/633

## Proposed changes:

* Moved the `nova.php` content across to the Classic Theme Helper package, as well as associated Nova CSS and JS files.
* Changed how assets were enqueued in the main Nova_Restaurant class, to use the Assets class - consistent with other files in the Classic Theme Helper package.
* Resolved some Phan-flagged errors, suppressed others where it made sense to.
* I have tested requiring the package to make sure that where possible no additional changes should be needed in the package before requiring, in the next PR.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/633
pfwV0U-ke-p2


## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* The package isn't being used yet, but as the build process is changing the location of the JS files in particular, a code review to check the enqueued JS file paths would make sense.
* If you do generally want to test the feature to make sure nothing is being added via the package, you would need to install and activate a theme that supports Restaurant Menus such as Confit.
  *  Add several new menu items from the new Food Menu in the admin menu.
  * Add some new sections, and add menu items to specific sections (see the Menu Sections area in the menu editor). Ensure some sections are parents to other sections
  * Attempt adding multiple menu items at once using the 'Add Many Items' menu option.
  * Using quick edit from the main menu list for specific item, change the sections they belong to in some cases.
  * Try dragging and dropping menu items to different positions and sections.
  * Click save new order, everything should save correctly.
* If testing, make sure you test with the changes applied for both the Jetpack Mu WPcom and Jetpack (so on Simple using both generated commands below) for WoA and Simple sites, and just Jetpack for self-hosted.
